### PR TITLE
runfix: Update quotes on visible messages (WEBAPP-5712)

### PR DESCRIPTION
**Pull request by AndyLnd**

### CODE REVIEW

File: app/script/conversation/ConversationRepository.js [hunk 1]
```diff
@@ -3491,6 +3491,20 @@ z.conversation.ConversationRepository = class ConversationRepository {
         if (conversationEntity && messageEntity) {
           const replacedEntity = conversationEntity.add_message(messageEntity, true);
           if (replacedEntity) {
+            const messages = conversationEntity.messages_unordered();
+
+            const updatedMessages = messages.map(message => {
+              const hasEditedQuote =
+                message.quote && message.quote() && message.quote().messageId === replacedEntity.id;
```
Comment (ffflorian) [inline app/script/conversation/ConversationRepository.js:3498]: A `Message` always has `message.quote`, so we don't need to check it.

```suggestion
                message.quote() && message.quote().messageId === replacedEntity.id;
```

Comment (atomrc) [reply app/script/conversation/ConversationRepository.js:3498]: I believe the check is legit. We should not remove it

File: app/script/conversation/ConversationRepository.js [hunk 2]
```diff
@@ -3491,6 +3491,20 @@ z.conversation.ConversationRepository = class ConversationRepository {
         if (conversationEntity && messageEntity) {
           const replacedEntity = conversationEntity.add_message(messageEntity, true);
           if (replacedEntity) {
+            const messages = conversationEntity.messages_unordered();
```
Comment (atomrc) [inline app/script/conversation/ConversationRepository.js:3494]: Possible optimization, you can use the ordered messages list (namely `messages` I believe) and do stop when messages are before the timestamp of the edited message

File: app/script/conversation/ConversationRepository.js (hunk 1 above)
Comment (ffflorian) [reply app/script/conversation/ConversationRepository.js:3498]: :+1:

**Pull request merged**